Windows 10 with paegent as ssh-agent cause a crash

### Operating system

Windows 10

### Kluctl Version

2.16.1

### Kubernetes Version

all

### kubectl Version

all

### Bug description

Using kluctl on Windows 10 with paegent as ssh agent causes a crash with the following stack trace:

```
panic: Failed to find GetCurrentThreadId procedure in user32.dll: The specified procedure could not be found.

goroutine 29 [running]:
golang.org/x/sys/windows.(*LazyProc).mustFind(...)
        /home/runner/go/pkg/mod/golang.org/x/sys@v0.1.0/windows/dll_windows.go:325
golang.org/x/sys/windows.(*LazyProc).Call(0xc000839c50, {0x0, 0x0, 0x0})
        /home/runner/go/pkg/mod/golang.org/x/sys@v0.1.0/windows/dll_windows.go:347 +0x65
github.com/xanzy/ssh-agent.winAPI.func1({0x0?, 0x78494a?, 0xc000ac6ec8?})
        /home/runner/go/pkg/mod/github.com/xanzy/ssh-agent@v0.3.2/pageant_windows.go:75 +0x2e
github.com/xanzy/ssh-agent.query({0xc000d86848, 0x5, 0x482ea0?})
        /home/runner/go/pkg/mod/github.com/xanzy/ssh-agent@v0.3.2/pageant_windows.go:100 +0x126
github.com/xanzy/ssh-agent.(*conn).Write(0xc000a2a6e0, {0xc000d86848?, 0x5, 0xc000ac71b0?})
        /home/runner/go/pkg/mod/github.com/xanzy/ssh-agent@v0.3.2/sshagent_windows.go:82 +0xe7
golang.org/x/crypto/ssh/agent.(*client).callRaw(0xc00011ec48, {0xc000ac7273, 0x1, 0x248cc78?})
        /home/runner/go/pkg/mod/golang.org/x/crypto@v0.1.0/ssh/agent/client.go:344 +0x15d
golang.org/x/crypto/ssh/agent.(*client).call(0x20b6f68?, {0xc000ac7273?, 0xc000ac72a8?, 0x78b033?})
        /home/runner/go/pkg/mod/golang.org/x/crypto@v0.1.0/ssh/agent/client.go:323 +0x25
golang.org/x/crypto/ssh/agent.(*client).List(0x21b57454260?)
        /home/runner/go/pkg/mod/golang.org/x/crypto@v0.1.0/ssh/agent/client.go:405 +0x34
golang.org/x/crypto/ssh/agent.(*client).Signers(0xc00011ec48)
        /home/runner/go/pkg/mod/golang.org/x/crypto@v0.1.0/ssh/agent/client.go:750 +0x33
github.com/kluctl/kluctl/v2/pkg/git/auth.(*sshDefaultIdentityAndAgent).addAgentIdentities(0xc0005b6190, {{{0xc0005aafc0, 0x3}, {0x0, 0x0}, 0xc00049fbc0, {0xc0005aafca, 0x1b}, {0xc0005aafe5, 0x1b}, ...}})
        /home/runner/work/kluctl/kluctl/pkg/git/auth/ssh_auth_provider.go:93 +0x10e
github.com/kluctl/kluctl/v2/pkg/git/auth.(*GitSshAuthProvider).BuildAuth(0xc000334120, {0x3cd2dd0?, 0xc0003a42a0}, {{{0xc0005aafc0, 0x3}, {0x0, 0x0}, 0xc00049fbc0, {0xc0005aafca, 0x1b}, ...}})
        /home/runner/work/kluctl/kluctl/pkg/git/auth/ssh_auth_provider.go:119 +0x238
github.com/kluctl/kluctl/v2/pkg/git/auth.(*GitAuthProviders).BuildAuth(0x886a7e?, {0x3cd2dd0, 0xc0003a42a0}, {{{0xc0005aafc0, 0x3}, {0x0, 0x0}, 0xc00049fbc0, {0xc0005aafca, 0x1b}, ...}})
        /home/runner/work/kluctl/kluctl/pkg/git/auth/auth_provider.go:36 +0x158
github.com/kluctl/kluctl/v2/pkg/git.(*MirroredGitRepo).update(0xc0006802d0, 0xc0008e42a0?, {0xc000d22500?, 0x7?})
        /home/runner/work/kluctl/kluctl/pkg/git/mirrored_repo.go:205 +0xd8
github.com/kluctl/kluctl/v2/pkg/git.(*MirroredGitRepo).cloneOrUpdate(0xc0006802d0, 0xc0003a42a0?)
        /home/runner/work/kluctl/kluctl/pkg/git/mirrored_repo.go:317 +0x297
github.com/kluctl/kluctl/v2/pkg/git.(*MirroredGitRepo).Update(0xc0006802d0)
        /home/runner/work/kluctl/kluctl/pkg/git/mirrored_repo.go:341 +0xbd
github.com/kluctl/kluctl/v2/pkg/git/repocache.(*CacheEntry).Update(0xc0005c1640)
        /home/runner/work/kluctl/kluctl/pkg/git/repocache/cache.go:121 +0x15f
github.com/kluctl/kluctl/v2/pkg/git/repocache.(*GitRepoCache).GetEntry(0xc0005b00e0, {{{0xc0005aafc0, 0x3}, {0x0, 0x0}, 0xc00049fbc0, {0xc0005aafca, 0x1b}, {0xc0005aafe5, 0x1b}, ...}})
        /home/runner/work/kluctl/kluctl/pkg/git/repocache/cache.go:100 +0x24b
github.com/kluctl/kluctl/v2/pkg/kluctl_project.(*LoadedKluctlProject).updateGitCaches.func2.1()
        /home/runner/work/kluctl/kluctl/pkg/kluctl_project/git.go:27 +0xb8
created by github.com/kluctl/kluctl/v2/pkg/kluctl_project.(*LoadedKluctlProject).updateGitCaches.func2
        /home/runner/work/kluctl/kluctl/pkg/kluctl_project/git.go:24 +0x11b
```

### Steps to reproduce

_No response_

### Relevant log output

_No response_

--- a/go.mod
+++ b/go.mod
@@ -37,7 +37,7 @@ require (
 	github.com/spf13/viper v1.13.0
 	github.com/stretchr/testify v1.8.0
 	github.com/whilp/git-urls v1.0.0
-	github.com/xanzy/ssh-agent v0.3.2
+	github.com/xanzy/ssh-agent v0.3.3-0.20220920102508-0fa644ba07f4
 	golang.org/x/crypto v0.1.0
 	golang.org/x/net v0.1.0
 	golang.org/x/sync v0.1.0

